macros/drop_dev_and_sinter_schemas.sql

```diff
@@ -0,0 +1,57 @@
+{% macro drop_dev_schemas(schema_ilike = ['dbt%', 'sinter%', 'dev%'], dry_run = true, exclude = '') %} 
+
+{% set get_dev_sql %}
+
+    select 
+        distinct schemaname
+    from pg_catalog.pg_tables 
+    where 
+    (
+        (
+        
+        {% for schema in schema_ilike %} 
+
+        schemaname ilike '{{schema}}' {{'or' if not loop.last}} 
+
+        {% endfor %}
+        
+    )
+        and 
+        (
+            
+        
+        {% for schema in schema_ilike %} 
+        
+        schemaname not ilike '{{ exclude }}' {{'or' if not loop.last}} 
+        
+        {% endfor %}
+    
+    )
+        
+    )
+{% endset %}
+
+
+{% set dev_schemas = dbt_utils.get_query_results_as_dict(get_dev_sql) %}
+
+{% for schema_name in dev_schemas['schemaname'] %} 
+
+    {% set drop_schema -%} 
+    
+        drop schema {{schema_name}} cascade;
+
+    {%- endset %}
+    
+    {% if dry_run %} 
+    
+        {{ log(drop_schema, info=True) }} 
```

Comment:
We might want some logging above here — people might get confused if they just see drop statements being logged. Something like "these are the statements that would have been run" (but not those words exactly).

One way to do this is to add the `drop` statements to an array first, and then have the `if` logic afterwards
```
{% set drop_statements = [] %}
{% for schema_name in dev_schemas['schemaname'] %} 
{% do drop_statements.append("drop schema " ~ schemaname ~ " cascade;"
{% endfor %}

{{ log("Drop statements to be run:", info=True) }}
{{ log(drop_statements | join(\n), info=True) }}

{% if dry_run %}
{{ log("dry run -- no statements executed") }}
{% else %}
{{ log("executing drop statements"), info=True }}
{% do run_query(drop_statements | join(\n), info=True) %}
{% endif %}

```

None of that code is tested, and there's probably some improvements to make!

Reply:
I think thats a great idea. Ill make these changes soon!